react/src/lib/EventOverlay/index.js

```diff
@@ -315,53 +301,56 @@ class EventOverlay extends React.Component {
       bottom: targetNode.offsetHeight,
       left: 0,
       middle: targetNode.offsetWidth / 2,
-      right: targetNode.offsetWidth
+      right: targetNode.offsetWidth,
     };
-  }
+  };
 
   handleAllowClickAway = e => {
     if (!this.props.isOpen) return;
+    const eventTarget = eventPath(e)[0];
     const anchorNode = ReactDOM.findDOMNode(this.props.anchorNode);
 
     return (
-      this.container
-        && !anchorNode.contains(e.target)
-        && !ReactDOM.findDOMNode(this.container).contains(e.target)
-        && this.handleClickAway(e)
+      this.container &&
+      !anchorNode.contains(eventTarget) &&
+      !ReactDOM.findDOMNode(this.container).contains(eventTarget) &&
+      this.handleClickAway(e)
     );
-  }
+  };
 
   handleClickAway = e => {
     const { close } = this.props;
 
     this.focusOnAnchorNode();
     close && close(e);
-  }
+  };
 
   handleCloseOnClick = e => {
     if (!this.props.isOpen) return;
+    const eventTarget = eventPath(e)[0];
```

Comment:
@yananym Look here for the applied changes (not just formatting), where `eventTarget` is a result of the polyfill, then checked in the `findDOMNode` below